internal/otelcollector/config/metric/agent/receivers_test.go

```diff
@@ -231,3 +266,107 @@ func TestReceivers(t *testing.T) {
+	metricsToDrop := K8sClusterMetricsToDrop{}
+
+	//nolint:dupl // repeating the code as we want to test the metrics are disabled correctly
+	defaultMetricsToDrop := &K8sClusterDefaultMetricsToDrop{
+		K8sContainerStorageRequest:          MetricConfig{Enabled: false},
+		K8sContainerStorageLimit:            MetricConfig{Enabled: false},
+		K8sContainerEphemeralStorageRequest: MetricConfig{Enabled: false},
+		K8sContainerEphemeralStorageLimit:   MetricConfig{Enabled: false},
+		K8sContainerRestarts:                MetricConfig{Enabled: false},
+		K8sContainerReady:                   MetricConfig{Enabled: false},
+		K8sNamespacePhase:                   MetricConfig{Enabled: false},
+		K8sHPACurrentReplicas:               MetricConfig{Enabled: false},
+		K8sHPADesiredReplicas:               MetricConfig{Enabled: false},
+		K8sHPAMinReplicas:                   MetricConfig{Enabled: false},
+		K8sHPAMaxReplicas:                   MetricConfig{Enabled: false},
+		K8sReplicaSetAvailable:              MetricConfig{Enabled: false},
+		K8sReplicaSetDesired:                MetricConfig{Enabled: false},
+		K8sReplicationControllerAvailable:   MetricConfig{Enabled: false},
+		K8sReplicationControllerDesired:     MetricConfig{Enabled: false},
+		K8sResourceQuotaHardLimit:           MetricConfig{Enabled: false},
+		K8sResourceQuotaUsed:                MetricConfig{Enabled: false},
+	}
+	podMetricsToDrop := &K8sClusterPodMetricsToDrop{
+		K8sPodPhase: MetricConfig{false},
+	}
+	containerMetricsToDrop := &K8sClusterContainerMetricsToDrop{
+		K8sContainerCPURequest:    MetricConfig{false},
+		K8sContainerCPULimit:      MetricConfig{false},
+		K8sContainerMemoryRequest: MetricConfig{false},
+		K8sContainerMemoryLimit:   MetricConfig{false},
+	}
+
+	statefulMetricsToDrop := &K8sClusterStatefulSetMetricsToDrop{
+		K8sStatefulSetCurrentPods: MetricConfig{false},
+		K8sStatefulSetDesiredPods: MetricConfig{false},
+		K8sStatefulSetReadyPods:   MetricConfig{false},
+		K8sStatefulSetUpdatedPods: MetricConfig{false},
+	}
+	jobMetricsToDrop := &K8sClusterJobMetricsToDrop{
+		K8sJobActivePods:            MetricConfig{false},
+		K8sJobDesiredSuccessfulPods: MetricConfig{false},
+		K8sJobFailedPods:            MetricConfig{false},
+		K8sJobMaxParallelPods:       MetricConfig{false},
+		K8sJobSuccessfulPods:        MetricConfig{false},
+	}
+	deploymentMetricsToDrop := &K8sClusterDeploymentMetricsToDrop{
+		K8sDeploymentAvailable: MetricConfig{false},
+		K8sDeploymentDesired:   MetricConfig{false},
+	}
+	daemonSetMetricsToDrop := &K8sClusterDaemonSetMetricsToDrop{
+		K8sDaemonSetCurrentScheduledNodes: MetricConfig{false},
+		K8sDaemonSetDesiredScheduledNodes: MetricConfig{false},
+		K8sDaemonSetMisscheduledNodes:     MetricConfig{false},
+		K8sDaemonSetReadyNodes:            MetricConfig{false},
+	}
+
+	if resourceEnabled == "default" {
+		metricsToDrop.K8sClusterDefaultMetricsToDrop = defaultMetricsToDrop
```

Comment:
setting the default metrics to drop can be done only once before all of these checks, as they will always be dropped and no need to repeat this statement inside every check